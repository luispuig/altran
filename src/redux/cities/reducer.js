// @flow
import {
  CITY_UPDATE__REQUEST,
  CITY_UPDATE__FAIL,
  CITY_UPDATE__SUCCESS
} from "./actions";

import type { _City, _ActionPayload } from "./types";

const initialData = { temperature: 0, state: "request", error: undefined };
const initialState: Array<_City> = [
  { id: 3871336, name: "Santiago", ...initialData },
  { id: 3435910, name: "Buenos Aires", ...initialData },
  { id: 3936456, name: "Lima", ...initialData },
  { id: 3448439, name: "Sao Paulo", ...initialData }
];

const reducer = (
  state: Array<_City> = initialState,
  action: _ActionPayload
) => {
  switch (action.type) {
    case CITY_UPDATE__FAIL:
      /* Update Fail fields */
      if (action.payload.error === undefined) return state;
      return state.map(item =>
        item.id === action.payload.id
          ? { ...item, state: "fail", error: action.payload.error }
          : item
      );

    case CITY_UPDATE__SUCCESS:
      /* Update Success fields */
      return state.map(item =>
        item.id === action.payload.id
          ? {
              ...item,
              state: "success",
              temperature: action.payload.temperature
            }
          : item
      );
    default:
      return state;
  }
};

export default reducer;
