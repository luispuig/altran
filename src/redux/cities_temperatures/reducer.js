// @flow

import {
  CITY_TEMPERATURE__ADD
} from "./actions";

import type { _Temperature, _ActionPayload} from "./types"

const initialState:Array<_Temperature> = [];

const reducer = (state:Array<_Temperature> = initialState, action:_ActionPayload) => {
  switch (action.type) {
    case CITY_TEMPERATURE__ADD:
      /* Add New Temperature to the batch */
      return [action.payload, ...state]; // Last value first

    default:
      return state;
  }
};

export default reducer;
