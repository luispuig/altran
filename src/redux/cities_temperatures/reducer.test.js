import {
  CITY_TEMPERATURE__ADD
} from "./actions";
import reducer from "./reducer";

it('Reducer other action', async () => {

  const action = {
    type: "NO-ACTION"
  };
  const state = reducer(undefined, action);
  expect(state).toMatchSnapshot();

});

it('Reducer other action: CITY_TEMPERATURE__ADD', async () => {

  const action = {
    type: CITY_TEMPERATURE__ADD,
    payload: { id: 3871336, temperature: 25 }
  };
  const state = reducer(undefined, action);
  expect(state).toMatchSnapshot();

});
