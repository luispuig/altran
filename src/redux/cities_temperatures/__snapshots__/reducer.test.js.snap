// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducer other action 1`] = `Array []`;

exports[`Reducer other action: CITY_TEMPERATURE__ADD 1`] = `
Array [
  Object {
    "id": 3871336,
    "temperature": 25,
  },
]
`;
